github/resource_github_repository_file.go

```diff
@@ -0,0 +1,350 @@
+package github
+
+import (
+	"context"
+	"log"
+	"net/http"
+	"strings"
+
+	"fmt"
+
+	"github.com/google/go-github/v28/github"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceGithubRepositoryFile() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGithubRepositoryFileCreate,
+		Read:   resourceGithubRepositoryFileRead,
+		Update: resourceGithubRepositoryFileUpdate,
+		Delete: resourceGithubRepositoryFileDelete,
+		Importer: &schema.ResourceImporter{
+			State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				parts := strings.Split(d.Id(), ":")
+				branch := "master"
+
+				if len(parts) > 2 {
+					return nil, fmt.Errorf("Invalid ID specified. Supplied ID must be written as <repository>/<file path> (when branch is \"master\") or <repository>/<file path>:<branch>")
+				}
+
+				if len(parts) == 2 {
+					branch = parts[1]
+				}
+
+				client := meta.(*Organization).client
+				org := meta.(*Organization).name
+				repo, file := splitRepoFilePath(parts[0])
+				if err := checkRepositoryFileExists(client, org, repo, file, branch); err != nil {
+					return nil, err
+				}
+
+				d.SetId(fmt.Sprintf("%s/%s", repo, file))
+				d.Set("branch", branch)
+
+				return []*schema.ResourceData{d}, nil
+			},
+		},
+
+		Schema: map[string]*schema.Schema{
+			"repository": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The repository name",
+			},
+			"file": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The file path to manage",
+			},
+			"content": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The file's content",
+			},
+			"branch": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The branch name, defaults to \"master\"",
+				Default:     "master",
+			},
+			"commit_message": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "The commit message when creating or updating the file",
+			},
+			"commit_author": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "The commit author name, defaults to the authenticated user's name",
+			},
+			"commit_email": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "The commit author email address, defaults to the authenticated user's email address",
+			},
+			"sha": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The blob SHA of the file",
+			},
+		},
+	}
+}
+
+func resourceGithubRepositoryFileOptions(d *schema.ResourceData) (*github.RepositoryContentFileOptions, error) {
```

Comment:
Nicely done here, good feedback to the user if they've configured the resource improperly.